### PR TITLE
Implement analytics-compat package

### PR DESCRIPTION
**Pull request by hsubox76**
Create analytics-compat package wrapping analytics-exp. Did not add to `firebase` package - doesn't seem like most compat packages have been added to it yet.

### CODE REVIEW

File: packages-exp/analytics-compat/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { firebase } from '@firebase/app-compat';
+import { _FirebaseNamespace } from '@firebase/app-types/private';
```
Comment (Feiyang1) [inline packages-exp/analytics-compat/src/index.ts:19]: FYI, we may have merge conflict with https://github.com/firebase/firebase-js-sdk/pull/4461 depending on which one got merged first.

File: packages-exp/analytics-compat/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    .getImmediate({
+      identifier: regionOrCustomDomain
+    });
+
+  return new AnalyticsService(app as FirebaseApp, analyticsServiceExp);
+};
+
+export function registerAnalytics(): void {
+  const namespaceExports = {
+    // no-inline
```
Comment (Feiyang1) [inline packages-exp/analytics-compat/src/index.ts:61]: what does the comment mean?

File: packages/analytics-types/index.d.ts [hunk 3]
```diff
@@ -40,13 +40,421 @@ export interface FirebaseAnalytics {
...
    * app instance on this device.
-   * List of official event parameters can be found in
+   * List of recommended event parameters can be found in
    * {@link https://developers.google.com/gtagjs/reference/event
    * the gtag.js reference documentation}.
    */
   logEvent(
-    eventName: EventNameString,
-    eventParams?: EventParams,
+    eventName: 'add_payment_info',
```
Comment (Feiyang1) [inline packages/analytics-types/index.d.ts:48]: Did you miss the overloads before?

File: packages/analytics/src/factory.ts [hunk 4]
```diff
@@ -226,7 +229,11 @@ export function factory(
     app,
     // Public methods return void for API simplicity and to better match gtag,
     // while internal implementations return promises.
-    logEvent: (eventName, eventParams, options) => {
+    logEvent: (
+      eventName: string,
```
Comment (Feiyang1) [inline packages/analytics/src/factory.ts:233]: Is it necessary to type them explicitly? Are they not inferred by `FirebaseAnalyticsInternal ` ?

File: packages/firebase/index.d.ts [hunk 5]
```diff
@@ -4479,7 +4479,7 @@ declare namespace firebase.analytics {
      */
     logEvent(
       eventName: 'add_shipping_info',
-      eventParams: {
+      eventParams?: {
```
Comment (Feiyang1) [inline packages/firebase/index.d.ts:4482]: These changes are unrelated to `-compat` and should have a changeset. We should create a separate PR for them.

**Review by Feiyang1 — CHANGES_REQUESTED** [review]

File: packages/analytics-types/index.d.ts (hunk 3 above)
Comment (hsubox76) [reply packages/analytics-types/index.d.ts:48]: It didn't seem necessary to put them in analytics-types because it's only for user convenience, so it was only in firebase/index.d.ts. But analytics-compat won't have access to firebase/index.d.ts.

File: packages/analytics/src/factory.ts (hunk 4 above)
Comment (hsubox76) [reply packages/analytics/src/factory.ts:233]: It used to be fine but after adding the logEvent overloads to analytics-types now it complains if the params aren't typed?

File: packages/firebase/index.d.ts (hunk 5 above)
Comment (hsubox76) [reply packages/firebase/index.d.ts:4482]: Done: https://github.com/firebase/firebase-js-sdk/pull/4469

File: packages-exp/analytics-compat/src/index.ts (hunk 2 above)
Comment (hsubox76) [reply packages-exp/analytics-compat/src/index.ts:61]: It's left over from copying from another file. I see it in functions and storage and I think it might have been an annotation for Prettier to keep the following line on its own line even when there is only one item. But it doesn't seem to do anything now even in files where there's only one item. Removed.

File: packages-exp/analytics-compat/src/constants.ts [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Officially recommended event names for gtag.js
+ * Any other string is also allowed.
+ *
+ * @public
```
[the remark this answers is not in the document]
Comment (hsubox76) [reply packages-exp/analytics-compat/src/constants.ts:22]: Removed.

**Review by Feiyang1 — APPROVED** [review]
